UtilityBeltTests/UtilityBeltDataTests/Protocols/CoreDataManagerTesting.swift

```diff
@@ -0,0 +1,82 @@
+// Copyright © 2020 SpotHero, Inc. All rights reserved.
+
+import CoreData
+import Foundation
+@testable import UtilityBeltData
+import XCTest
+
+protocol CoreDataManagerTesting: XCTestCase {
+    var storeType: NSPersistentContainer.StoreType { get }
+}
+
+extension CoreDataManagerTesting {
+    func initialize(file: StaticString = #file, line: UInt = #line) {
+        CoreDataManager.defaultContainer = .mocked(name: "UtilityBeltData", storeType: self.storeType)
+
+        do {
+            try self.loadData()
+        } catch {
+            XCTFail(error.localizedDescription, file: file, line: line)
+        }
+    }
+
+    func count(file: StaticString = #file, line: UInt = #line) {
+        self.measureAndCatch {
+            let userCount = try CoreDataManager.default.count(of: User.self)
+
+            XCTAssertEqual(userCount, 4, file: file, line: line)
+        }
+    }
+
+    func delete(file: StaticString = #file, line: UInt = #line) {
+        XCTAssertEqual(self.storeType, .sqlite)
```

Comment:
not sure I understand whats going on with this assertion?   Why does only delete need to require `.sqlite`?

Reply:
This is commented in the `deleteAll` function on `CoreDataManger`:

`Batch delete requests are only compatible with the SQLite store type.`

I added the source here:

https://developer.apple.com/library/archive/featuredarticles/CoreData_Batch_Guide/BatchDeletes/BatchDeletes.html